### PR TITLE
nim: cross-compilation fixes

### PR DESCRIPTION
**Pull request by ghost**
Import with callPackages, use buildPackages for building a cross-compiler, cleanup nim.cfg so that it does not conflict with our configuration.

###### Motivation for this change

Cross-compilation doesn't work right now, the wrapper takes a Nim compiler built for the targetPlatform.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [.] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/nim/default.nix [hunk 1]
```diff
@@ -83,111 +70,120 @@ let
...
+    pname = "nim-bootstrap";
+    version = "0.20.0";
 
-    bootstrap = stdenv.mkDerivation rec {
-      pname = "nim-bootstrap";
-      version = "0.20.0";
+    src = fetchgit {
+      # A Git checkout is much smaller than a GitHub tarball.
+      url = "https://github.com/nim-lang/csources.git";
+      rev = "v" + version;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/nim/default.nix:80]:
```suggestion
      rev = "v${version}";
```

File: pkgs/development/compilers/nim/default.nix [hunk 2]
```diff
@@ -83,111 +70,120 @@ let
...
-      src = fetchFromGitHub {
-        owner = "nim-lang";
-        repo = "nimble";
-        rev = "v" + version;
-        sha256 = "1idb4r0kjbqv16r6bgmxlr13w2vgq5332hmnc8pjbxiyfwm075x8";
-      };
+    src = fetchFromGitHub {
+      owner = "nim-lang";
+      repo = "nimble";
+      rev = "v" + version;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/nim/default.nix:163]:
```suggestion
      rev = "v${version}";
```

**Review by SuperSandro2000 — COMMENTED** [review]
Small formatting nit and you might want to set ``strictDeps = true;`` to easier catch regressions for future updates.

**Comment by ehmry** [comment]
@SuperSandro2000 Yes, strictDeps is what I want, thanks.

**Comment by ehmry** [comment]
I'm having problems getting cross-complation to work were the build platform is GCC and the target platform is Clang.

File: pkgs/development/compilers/nim/default.nix [hunk 3]
```diff
@@ -83,197 +71,251 @@ let
...
+        WTF
+
+        runHook postBuild
+      '';
+
     wrapperArgs = [
-      "--prefix PATH : ${lib.makeBinPath [ stdenv.cc gdb ]}:${
+      "--prefix PATH : ${lib.makeBinPath [ buildPackages.gdb ]}:${
         placeholder "out"
       }/bin"
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/nim/default.nix:261]:
```suggestion
      "--prefix PATH : ${lib.makeBinPath [ buildPackages.gdb ]}:${placeholder "out"}/bin"
```

File: pkgs/development/compilers/nim/default.nix [hunk 4]
```diff
@@ -83,197 +71,251 @@ let
...
     dontConfigure = true;
 
+    buildPhase =
+      # Configure the Nim compiler to use $CC and $CXX as backends
+      # The compiler is configured by two configuration files, each with
+      # a different DSL. The order of evaluation matters and that order
+      # is not documented, so duplicate the configuration across both files.
+      ''
+        runHook preBuild
+        cat >> config/config.nims << WTF
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/nim/default.nix:218]: I would prefer if EOF for heredocs are used. That is more common and for everyone easy to spot.

**Comment by SuperSandro2000** [comment]
> I'm having problems getting cross-complation to work where the build platform is GCC and the target platform is Clang.

If you need more build power or some testing and some limited help on darwin just hit me up.

**Pull request merged**